### PR TITLE
[FIX] project: fix access error when sharing project link

### PR DESCRIPTION
**Pull request by mjvi-odoo**
Steps to reproduce:
- Open the project kanban card.
- Click "Share" and copy the link.
- Open the link in an incognito window.

Issue:
- Access is not getting for search_count

Solution:
- Give the sudo access.

task-3884334

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/project/controllers/portal.py [hunk 1]
```diff
@@ -410,7 +410,7 @@ def _prepare_tasks_values(self, page, date_begin, date_end, sortby, search, sear
 
         Task = request.env['project.task']
         milestone_domain = AND([domain, [('allow_milestones', '=', True)], [('milestone_id', '!=', False)]])
-        milestones_allowed = Task.search_count(milestone_domain, limit=1) == 1
+        milestones_allowed = Task.sudo().search_count(milestone_domain, limit=1) == 1
```
Comment (damrOdoo) [inline addons/project/controllers/portal.py:413]: Adding a sudo here is not a good idea. The goal of the search_count is to check if out of the tasks that the user has access to, some have the 'milestone_allowed' field set to true. Using a sudo will check for all the task in the db, so its a change in the behavior. Sadly, I dont really have a better idea on how to solve the access error while keeping the current behavior. I suppose we could add a condition on the domain with the collaborators of the tasks, but iirc it is quite costly to search on.
@xavierbol Do you have a better suggestion ? :angel:

**Review by damrOdoo — COMMENTED** [review]
Thanks for your fix, i left a comment. Sadly i disagree with your solution, but i dont have a better idea either, so i asked xavier for his input on this.

File: addons/project/controllers/portal.py (hunk 1 above)
Comment (xavierbol) [reply addons/project/controllers/portal.py:413]: @damrOdoo we could maybe move the code in L413-417 below the code at the line 423. 🤔 

What do you think? 🙂

Comment (damrOdoo) [reply addons/project/controllers/portal.py:413]: yup, good idea, lets go with that.

Comment (mjvi-odoo) [reply addons/project/controllers/portal.py]: @damrOdoo 
done with changes

**Comment by xavierbol** [comment]
LGTM, thanks for your work. 🙂 

robodoo r+

**Pull request closed without merging**